[14.0][MIG]purchase_requisition_auto_rfq

anddago78 opened the pull request:
This module adds a button on the purchase requisition form to create RFQs
using the suppliers from the products listed in the requisition.

MT-2584 @moduon @rafaelbn @Shide @yajo  please review :)

gurneyalex committed:
new addon purchase_requisition_auto_rfq

This addon adds a button to generate RFQ for the purchase requisition
using the product.supplierinfo in order to only have the lines corresponding to
the products sold by each supplier.

lepistone committed:
rephrase comments on tests

lepistone committed:
refactor tests to be more readable

Here I take advantage of the new API and the assertions from nose.tools
to refactor the tests and make them more readable. They are still green
and should be otherwise unchanged.

lepistone committed:
make method called by button return True

The method auto_rfq_from_suppliers returned a list of ids of created
RFQs. This list was used by the tests only. In production a method
called by a button should just return True in case of success.

Now the method returns True, and tests are adapted to figure out the
created rfqs from requisition.purchase_ids instead of using the return
value.

lepistone committed:
improve assertion: compare dict with dict

This way assert_equal gives a very nice failure singling out the
different keys and values in the dictionary.

This is to prepare a test on product quantities, still getting good
failures if for example only the quantities for one line are different.

lepistone committed:
improve tests to check quantities as well

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

yvaucher committed:
Add missing default oca icons

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

Jerther committed:
[12.0][MIG] purchase_requisition_auto_rfq

oca-travis committed:
[UPD] Update purchase_requisition_auto_rfq.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_requisition_auto_rfq
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_requisition_auto_rfq/

fcvalgar reviewed (approved):
LGTM, Thank you @anddago78

yajo commented:
/ocabot migration purchase_requisition_auto_rfq

Shide commented on purchase_requisition_auto_rfq/models/purchase_requisition.py:
```diff
@@ -0,0 +1,67 @@
+        return self._create_rfqs(seller_products)
+
+    def _get_sellers(self):
+        seller_products = defaultdict(set)
```
Can be done with 
```suggestion
        seller_products = defaultdict(self.env["product.product"])
```
and change below
```python
seller_products[seller.name.id] |= line.product_id
```

But in the end (`_remove_lines_without_official_supplier` function)  doesn't impacts too much and will be the same code.

Shide reviewed (approved):
LGTM. Code and Functional review.

Commit:
[IMP] purchase_requisition_auto_rfq: pre-commit

yajo commented on purchase_requisition_auto_rfq/models/purchase_requisition.py:
```diff
@@ -58,7 +57,11 @@ def _post_products_without_supplier_message(self, products):
+            uid = self.user_id.id
+        else:
+            uid = self.env.user.id
+        ctx = {"default_requisition_id": self.id, "default_user_id": uid}
```
Please extract this change in a separate commit, because it is a new functionality.

Commit:
[MIG] purchase_requisition_auto_rfq: migration to 14.0

In newer versions, Odoo only logs 1 message for record creation. This is reflected in the changes in tests.

@moduon MT-2584

Co-authored-by: Jairo Llopis <973709+yajo@users.noreply.github.com>

anddago78 commented:
All comments attended

yajo commented:
This one seems ready to merge. @rafaelbn do you have permissions?

yajo reviewed (approved):
Reviewed new improvements.

rafaelbn reviewed (commented):
- [x] Just improve README a little bit but functionally 👍 
- [x] I think you can add Moduon as author in this case
- [x] I think you can add yourself @anddago78 as contributor AND maintainer

Thank you all!

yajo commented:
Please some merge? @OCA/procurement-purchase-maintainers

rafaelbn commented:
Are my comments attended @anddago78  ? https://github.com/OCA/purchase-workflow/pull/1817#pullrequestreview-1359869609

anddago78 commented:
> * [ ]  Just improve README a little bit but functionally +1
> * [ ]  I think you can add Moduon as author in this case
> * [ ]  I think you can add yourself @anddago78 as contributor AND maintainer
> 
> Thank you all!

All comments are attended!

HviorForgeFlow commented:
@anddago78 you must run pre-commit locally to fix pip setup installation when module name changes

anddago78 commented:
Thanks @HviorForgeFlow for your comment!

HviorForgeFlow commented:
/ocabot merge nobump
